cilium/CFP-39757-doc-improvement-plan.md

```diff
@@ -0,0 +1,39 @@
+# CFP-003: Template
+
+**SIG: Documentation** ([View all current SIGs](https://docs.cilium.io/en/stable/community/community/#all-sigs))
+
+**Begin Design Discussion:** 2025-05-27
+
+**Cilium Release:** TBD
+
+**Authors:** Peter Matulis <peter.matulis@isovalent.com>
+
+**Status:** Implementable
+
+## Summary
+
+Improve the Cilium documentation across the board.
+
+## Motivation
+
+Make the documentation better for all users of Cilium.
+
+## Goals
+
+Implement the following:
+
+* documentation suggestions - elements that should be considered for every software feature
+* documentation language - common terms that optimizes communication among contributors and reviewers
+* documentation types - a rational framework for how to think about documentation
+* formal structure - leading to consistency across the doc set
+* readiness level definitions (for Beta and Stable)
+* practical and detailed resources
+    * example documentation outlines
+    * templates that assist with the planning and writing of documentation
+    * guidelines on how certain files and directories should be structured
+
+## Proposal
+
+### Overview
+
+I wrote everything in this [Google document](https://docs.google.com/document/d/1S5tNARwnR4heBro2RwgcQjq4GFjb4l_LW8BAaaMUa94).
```

Comment:
To formally approve this PR, I'd want to see the content committed to this repository to ensure that the information remains accessible and that we've agreed to the same set of text. It's fine to change things afterwards / iterate on the CFP as we attempt to implement it, but it could introduce confusion if the agreed upon ideas are not stored in this tree.